### ISSUE TITLE
Poster hiding first story page video

### ISSUE THREAD

**Issue by gmajoulet** [ampproject/amphtml#26718]
On Android, the poster sometimes hides the first story page video. It is not removed as the video playback starts.

How things work for AMP:
1. `amp-video.buildCallback`
2. `amp-video.layoutCallback` (async, will resolve when video `loadedmetadata` triggers)
3. `firstLayoutCompleted` is called when `layoutCallback` resolves, and removes the poster on Android

On AMP Stories, the `<video>` element can be swapped in/out of the DOM. If it is swapped in between the steps (2) and (3), the `loadedmetadata` event on the swapped out `<video>` element will never trigger, and step (3) will never happen.

To repro: add a video on the first page of a story, and emulate on android device.

**Comment by gmajoulet** [comment]
> 1. amp-video.buildCallback
> 2. amp-video.layoutCallback (async, will resolve when video loadedmetadata triggers)
> 3. firstLayoutCompleted is called when layoutCallback resolves, and removes the poster on Android

We're submitting the following fix:

```
amp-story-page[active]:not(.i-amphtml-layout) amp-video.i-amphtml-poolbound {
  display: none !important;
}
```

The `.i-amphtml-poolbound` class is added by `amp-video.js` during the `buildCallback`, between the step 1 and 2.
`amp-story-page[active]:not(.i-amphtml-layout)` means that the active page `layoutCallback` did not complete yet. Part of the `layoutCallback` is getting the media pool ready to swap the `<video>` elements.

What this selector means is: if the `<amp-video>` is built before the `amp-story-page` replaced the `video` element through the media pool, delay the `layoutCallback` (do not do (2)) that would fail, to after the `video` is replaced.

**Comment by gmajoulet** [comment]
Closed by #26744
